docs/user-guide/api-mediation/certificate-configuration-scenarios.md

```diff
@@ -0,0 +1,252 @@
+# Certificate configuration scenarios
+
+As a system programmer, review the different scenarios for configuring Zowe for automatic certificate setup. Examples are provided for each scenario.
+
+Zowe servers require both a keystore to store the certificates as well as a truststore to validate of these certificates and any other certificates that are encountered.
+For use of Zowe on z/OS, keystore and truststore can either be unix file-based (PKCS12) or z/OS keyring-based. 
+Both the keystore and truststore can automatically be created by Zowe regardless of which storage type is used. Keystores and truststores can be populated either with certificates that the user chooses, or with self-signed certificates generated by Zowe. 
+This automation can be performed by defining and customizing the `zowe.setup.certificate` section of your Zowe YAML configuration.
```

Comment:
I find it intuitive to use `foo.bar.baz` format to quickly describe which section of the YAML is being described.
But, I have to assume nothing with our users. Should we hyperlink the heck out of these code blocks once or every time they appear like so? [`zowe.setup.certificate`](https://github.com/zowe/zowe-install-packaging/blob/v2.x/staging/example-zowe.yaml#L99) [[schema]](https://github.com/zowe/zowe-install-packaging/blob/v2.x/staging/schemas/zowe-yaml-schema.json#L163) [[syntax]](using-yaml.md)

We need to make such a "using-yaml.md" page to link... but basically, those 3 links would in theory completely contextualize a section of config, but would be tedious to do every single time; we'd probably forget.

Reply:
Does this mean linking to the yaml in all cases? I think a single link to the yaml would be sufficient, but also include the particular configuration per each scenario in the yaml example section. Perhaps Joe has these customized yaml examples as per scenario...

Reply:
example-zowe.yaml is currently our 1 and only YAML file. We're going to split it into several to allow the doc to more easily say "grab this whole file" or "see this whole file", but that's coming in a few weeks/months.

Almost all of the `foo.bar.baz` stuff can be pinpointed in example-zowe.yaml today, so we COULD link to it every single place we do this, but again I ask do we want to as a rule? I think our users would like it more than they dislike the clutter of doing that, but just my opinion...

Schema on the other hand, there really is only 1 for that purpose. Schema describes whats valid beyond examples. The example may show 1 thing, but the schema can tell you there's 5 alternatives.

As for "Syntax", I think we need a page much like this which tells users how to use YAML and what we mean by `this.that.other` https://www.ibm.com/docs/en/umsfz/1.2.0?topic=references-yaml-format

Reply:
I think this would provide a high degree of detail and could serve users well for now until the "grab this whole file" solution is developed... hopefully sooner than later.

Reply:
My 2 cents ... I think we should follow one major approach how to help the users to understand our documents. 

One approach is to provide educational material, point the uninformed users there once from every dependent document, and later expect that they either learned the concepts or at least can navigate themselves to that place anytime they need. 

The other approach would be to not bet on the user's knowledge, and provide the links everywhere. 

The first one is IMHO cleaner and easier to maintain, while the later one would clutter the text with repetitive links without significant benefits for the user. I can imagine they'd open a new tab to the same document (yaml, schema,...) every time they see a link, or even worse, may open it in the same tab replacing the main document, and lose their focus.